gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -1494,19 +1500,44 @@ private static long getMaxRemainingResults(
     List<String> listedPrefixes = new ArrayList<>();
     String nextPageToken =
         listStorageObjectsAndPrefixesPage(listObject, listOptions, listedObjects, listedPrefixes);
+    List<GoogleCloudStorageItemInfo> objectInfos =
+        getGoogleCloudStorageItemInfos(
+            bucketName, objectNamePrefix, listOptions, listedPrefixes, listedObjects);
+    return new ListPage<>(objectInfos, nextPageToken);
+  }
+
+  private List<GoogleCloudStorageItemInfo> getGoogleCloudStorageItemInfos(
```

Comment:
Can this method be reused in the listObjectNames method above? I find core logic are pretty similar.

Reply:
While logic is similar, `listObjectNames` does not create `GoogleCloudStorageItemInfo` objects to improve performance so it's hard to reuse the same code between these 2 methods.